src/cmds/wc.c

```diff
@@ -0,0 +1,170 @@
+/**
+ * @file
+ * @brief wc-util
+ *
+ * @date 19.07.18
+ * @author Filipp Chubukov
+ */
+
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <mem/misc/pool.h>
+#include <stdbool.h>
+#include <util/log.h>
+#define OBJECTS_QUANTITY 0x10
+
+POOL_DEF(pool, int, OBJECTS_QUANTITY);
+
+static int wc_util(FILE* file_in, int *lines_in_file, int *words_in_file, 
+		int *bytes_in_file,int *max_line_in_file) {
+
+	int byte_walker;
+	int count_len;
+	bool  prev_symbol = false;
+	bool  next_symbol = false;
+	*lines_in_file = 0;
+        *words_in_file = 0;
+        *bytes_in_file = 0;
+        *max_line_in_file = 0;
+	
+	while ((byte_walker = fgetc(file_in)) != EOF) {
+		*bytes_in_file = *bytes_in_file + 1; 	
+		
+		if (byte_walker == '\n') { 
+            		*lines_in_file = *lines_in_file + 1;	
+
+			if (*max_line_in_file < count_len) {
+				*max_line_in_file = count_len;
+			}
+
+			count_len = 0;
+		} else { 
+			if (byte_walker == '\t') {
+				count_len = count_len + 8; 
+
+			} else {
+				count_len++;
+			}
+		}
+
+		next_symbol = !isspace(byte_walker)  ?  true  :  false; 
+     
+		if (next_symbol) {
+             		prev_symbol = true;
+          	}
+     
+          	if (prev_symbol && !next_symbol){
+                	prev_symbol = false;
+    			*words_in_file = *words_in_file + 1;
+          	}
+
+	}
+	return 0;
+}
+
+int main(int argc, char **argv) { 
+	char *file_name;
+	int opt, param_number;
+	param_number = 0;
+
+	static bool print_lines, print_words,print_bytes;
+	static bool print_linelength;
+	
+	print_lines = print_words = print_bytes = false;
+  	print_linelength = false;
+
+	if (argc == 0) {
```

Comment:
`argc` generally can not be zero if OS is working correctly. You can replace this condition and the following with one, like

```
if (argc < 2) {
    printf("No file name given.\n");
}
```

Reply:
this check was used in another command, so I decided that it was needed